### PR TITLE
Moves constants to `settings.py`; #993

### LINKED ISSUE #993: Move constants to `settings.py`
Currently constants are accessed as `getattr(settings, SETTING_NAME, SETTING_DEFAULT)`.

This leads to a lot of hardcoding of `SETTING_DEFAULT`, as well as unnecessarily long code that could be avoided if these constants and their default values were instead added to `settings.py`.

### PR DESCRIPTION
**Pull request by Carson-Tang**
Fixes #993

### CODE REVIEW

File: dmoj/settings.py [hunk 1]
```diff
@@ -312,7 +391,7 @@
 ENABLE_FTS = False
 
 # Bridged configuration
-BRIDGED_JUDGE_ADDRESS = [('localhost', 9999)]
+BRIDGED_JUDGE_ADDRESS = [('localhost', 8080)]
```
Comment (quantum5) [inline dmoj/settings.py]: Revert this.

File: django_ace/widgets.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 from django.forms.utils import flatatt
 from django.utils.safestring import mark_safe
 
-ACE_URL = getattr(settings, 'ACE_URL', '//cdnjs.cloudflare.com/ajax/libs/ace/1.1.3/ext-split.js')
```
Comment (Xyene) [inline django_ace/widgets.py]: Can inline this now.

File: judge/performance_points.py [hunk 3]
```diff
@@ -6,8 +6,8 @@
 from judge.models import Submission
 from judge.timezone import from_database_time
 
-PP_STEP = getattr(settings, 'DMOJ_PP_STEP', 0.95)
-PP_ENTRIES = getattr(settings, 'DMOJ_PP_ENTRIES', 100)
+PP_STEP = settings.DMOJ_PP_STEP
```
Comment (Xyene) [inline judge/performance_points.py]: Inline these.

File: judge/utils/camo.py [hunk 4]
```diff
@@ -37,9 +37,9 @@ def update_tree(self, doc):
                 obj.set('data', self.rewrite_url(obj.get('data')))
 
 
-if getattr(settings, 'DMOJ_CAMO_URL', None) and getattr(settings, 'DMOJ_CAMO_KEY', None):
-    client = CamoClient(settings.DMOJ_CAMO_URL, key=settings.DMOJ_CAMO_KEY,
-                        excluded=getattr(settings, 'DMOJ_CAMO_EXCLUDE', ()),
-                        https=getattr(settings, 'DMOJ_CAMO_HTTPS', False))
+if settings.DMOJ_CAMO_URL and settings.DMOJ_CAMO_KEY:
+    client = CamoClient(settings.DMOJ_CAMO_URL, key = settings.DMOJ_CAMO_KEY,
```
Comment (Xyene) [inline judge/utils/camo.py]: No spaces before/after `=`.

File: judge/utils/subscription.py [hunk 5]
```diff
@@ -5,4 +5,4 @@
 else:
     Subscription = None
 
-newsletter_id = None if Subscription is None else getattr(settings, 'NEWSLETTER_ID_ON_REGISTER', None)
+newsletter_id = None if Subscription is None else settings.NEWSLETTER_ID_ON_REGISTER
```
Comment (Xyene) [inline judge/utils/subscription.py]: Rename with `DMOJ_` prefix.

File: judge/widgets/select2.py [hunk 6]
```diff
@@ -92,9 +90,9 @@ def _get_media(self):
             https://docs.djangoproject.com/en/1.8/topics/forms/media/#media-as-a-dynamic-property
         """
         return forms.Media(
-            js=(getattr(settings, 'SELECT2_JS_URL', DEFAULT_SELECT2_JS),
+            js=(settings.SELECT2_JS_URL,
                 'django_select2.js'),
-            css={'screen': (getattr(settings, 'SELECT2_CSS_URL', DEFAULT_SELECT2_CSS),)}
+            css={'screen': (settings.SELECT2_CSS_URL)}
```
Comment (Xyene) [inline judge/widgets/select2.py]: The original was a tuple; this no longer is.

File: dmoj/settings.py [hunk 7]
```diff
@@ -32,8 +33,86 @@
...
+NOFOLLOW_EXCLUDED = set()
+
+TIMEZONE_BG = None
+TIMEZONE_MAP = None
+TIMEZONE_DETECT_BACKEND = None
+
+TERMS_OF_SERVICE_URL = None
+DEFAULT_USER_LANGUAGE = 'PY2'
+
+COMPRESS_ENABLED = False
```
Comment (Xyene) [inline dmoj/settings.py]: I guess this should still be `not DEBUG`.

File: dmoj/settings.py [hunk 8]
```diff
@@ -32,8 +33,86 @@
 SITE_ID = 1
 SITE_NAME = 'DMOJ'
 SITE_LONG_NAME = 'DMOJ: Modern Online Judge'
+SITE_ADMIN_EMAIL = False
+
+# Constants
+DMOJ_SSL = 0
```
Comment (Xyene) [inline dmoj/settings.py]: Add comment for what the possible values here are (https://github.com/DMOJ/site/pull/1079/files#diff-43f2be90fc113eb0e09a185fcfc97b69L20)

**Pull request closed without merging**